src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -762,13 +762,26 @@ class spell_dk_death_pact : public SpellScript
     {
         // Check if we have valid targets, otherwise skip spell casting here
         if (Player* player = GetCaster()->ToPlayer())
+        {
             for (Unit::ControlList::const_iterator itr = player->m_Controlled.begin(); itr != player->m_Controlled.end(); ++itr)
+            {
                 if (Creature* undeadPet = (*itr)->ToCreature())
+                {
                     if (undeadPet->IsAlive() &&
                         undeadPet->GetOwnerGUID() == player->GetGUID() &&
                         undeadPet->GetCreatureType() == CREATURE_TYPE_UNDEAD &&
                         undeadPet->IsWithinDist(player, 100.0f, false))
+                    {
+                        if (undeadPet->HasAura(33786) || undeadPet->HasAura(65859))
```

Comment:
Don't check by spell id, use `IsImmunedToSpell`

Reply:
well I tried some things with IsImmunedToSpell but didn't managed to compile, either I'm using wrong arguments either I shoud declare them first

Reply:
I tried many combinations with IsImmunedToSpell but it doesn't accept my arguments and I didn't quite understood how this works, also did not found any valid example with this function used elsewhere.